library/spdm_responder_lib/libspdm_rsp_encap_response.c

```diff
@@ -124,6 +124,15 @@ static libspdm_return_t libspdm_process_encapsulated_response(
     bool need_continue;
     libspdm_encap_response_struct_t encap_response_struct;
 
+    /* the Requester delivers an encapsulated ERROR message, the Responder shall terminates the encapsulated request flow. */
+    if (encap_response != NULL) {
+        if (((const spdm_message_header_t *)encap_response)->request_response_code == SPDM_ERROR) {
```

Comment:
I dont think this line code is proper here.

It should be in a generic request_response_code checker/dispatcher.

Reply:
I have updated libspdm_handle_encap_error_response_main().